plugins/woocommerce/src/StoreApi/Utilities/SanitizationUtils.php

```diff
@@ -0,0 +1,27 @@
+<?php
+namespace Automattic\WooCommerce\StoreApi\Utilities;
+
+/**
+ * SanitizationUtils class.
+ * Helper class which sanitizes customer info.
+ */
+class SanitizationUtils {
+
+	/**
+	 * Runs wp_kses on an array. This function runs wp_kses on strings in the array and recurses into arrays.
+	 *
+	 * @param  array $array The array to run wp_kses on.
+	 * @return mixed       The array, all string keys will have been run through wp_kses.
+	 */
+	public function wp_kses_array( array $array ) {
+		foreach ( $array as $key => $value ) {
+			if ( is_array( $value ) ) {
+				$array[ $key ] = $this->wp_kses_array( $value );
+			}
+			if ( is_string( $value ) ) {
+				$array[ $key ] = wp_kses( $value, [] );
```

Comment:
Going through the rabbit hole of `wp_kses()` (heavily recommended), at some point it resorts to [`preg_replace_callback()`](https://www.php.net/manual/en/function.preg-replace-callback.php) which might return `null` on errors. By adding the suggested code above, we can check for `null` here (original `null` values would have been caught and skipped already) and handle this.
Not saying we need it, but would it be ok stripping stuff out and sending null in it's place? 🤔

Reply:
Check for null after the wp_kses has run? I think if the string is cleaned and something causes it to be null then it's fine to return null, something else went even more wrong.

Reply:
Fair enough, resolving and approving this 👍